tests/test_google_ads_field_exclusion_invalid.py

```diff
@@ -0,0 +1,245 @@
+"""Test tap field exclusions for invalid selection sets."""
+import re
+import random
+import pprint
+
+from tap_tester import menagerie, connections, runner
+
+from base import GoogleAdsBase
+
+
+class FieldExclusionInvalidGoogleAds(GoogleAdsBase):
+    """
+    Test tap's field exclusion logic with invalid selection for all streams
+    TODO Verify when given field selected, `fieldExclusions` fields in metadata are grayed out and cannot be selected (Manual case)
+    """
+
+    @staticmethod
+    def name():
+        return "tt_google_ads_field_exclusion_invalid"
+
+    def perform_exclusion_verification(self, field_exclusion_dict):
+        """
+        Verify for a pair of fields that if field_1 is in field_2's exclusion list then field_2 must be in field_1's exclusion list
+        """
+        error_dict = {}
+        for field, values in field_exclusion_dict.items():
+            if values != []:
+                for value in values:
+                    if value in field_exclusion_dict.keys():
+                        if field not in field_exclusion_dict[value]:
+                            if field not in error_dict.keys():
+                                error_dict[field] = [value]
+                            else:
+                                error_dict[field] += [value]
+
+        return error_dict
+
+    def random_field_gather(self, input_fields_with_exclusions):
```

Comment:
Now that these are two separate tests it may be better to keep this method in base under the Tap Specific section. Normally I like the "rule of 3" approach but for methods used in multiple tests for a given suite I think it makes sense.

Reply:
Ah I was too quick to write that previous comment, I forgot this is doing more than the other test.